### ISSUE TITLE
postdrop: warning: mail_queue_enter: create file maildrop/887233.343246: Permission denied

### ISSUE THREAD

**Issue by internethering** [systemd-cron/systemd-cron#75]
```
Jul 28 12:14:29 alpha.test mail_on_failure[343246]: postdrop: warning: mail_queue_enter: create file maildrop/887233.343246: Permission denied
Jul 28 12:14:29 alpha.test postfix/postdrop[343246]: warning: mail_queue_enter: create file maildrop/887233.343246: Permission denied
```

`postfix check` shows no output, the mail system is operation without any problems. how to debug mail_on_failure?

postfix: 3.6.1
systemd-cron: 1.5.17

**Comment by smcv** [comment]
I believe this is caused by the switch to `DynamicUser=yes`, which implies `NoNewPrivileges=yes`.

Postfix's `/usr/sbin/sendmail` relies on being able to run a setgid executable, `/usr/sbin/postdrop`, which has access to the `postdrop` group and therefore can write to `/var/spool/postfix/maildrop`. However, because systemd-cron's `mail_on_failure` is running with `PR_SET_NO_NEW_PRIVS`, it cannot gain privileges by running a setuid or setgid executable.

**Comment by a-detiste** [comment]
So there's no clear solution

**Comment by smcv** [comment]
I think the best solution might be to stop using `DynamicUser`, and instead use something like

```
[Service]
DynamicUser=no
User=_cron-failure
Group=systemd-journal
```

... and then install a `sysusers.d` snippet like `/usr/lib/sysusers.d/systemd-cron.conf` with something like

```
u _cron-failure -:systemd-journal - /nonexistent /usr/sbin/nologin
```

to create a non-dynamic `_cron-failure` system user to send the failure emails.

**Comment by mmokrejs** [comment]
> I think the best solution might be to stop using `DynamicUser`, and instead use something like
> 
> ```
> [Service]
> DynamicUser=no
> User=_cron-failure
> Group=systemd-journal
> ```

I ran into this issue too when migrating from OpenRC to systemd on Gentoo Linux. About what file are you speaking here?

**Comment by internethering** [comment]
I opened this bug 'n half of a year ago. May there someone from the devs, which can look after this?

**Comment by a-detiste** [comment]
Fixed by release v1.5.18
